### PR TITLE
Feature/safe test and extra tasks

### PR DESCRIPTION
**Pull request by AndrewTrow**
New tasks for safe test as well as additional commands that are pre-loaded in gitlab

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,14 +19,16 @@ The standard gradle pipeline is the simplest way to get up an running quickly. I
...
+| STANDARD_GRADLE_FLAGS   	|           | -s --no-daemon -PnoMavenLocal --refresh-dependencies --console=plain  (-PsafeTest)	| Default Gradle flags that will be appended to all Gradle commands (Will include -PsafeTest when set)                                                                                        |
+| DEV_REGEX                 |           | develop                                                                               | Branch(es) jobs will be run from when new commits are made. For example, if it's desired to run jobs from from `v2-develop` and `v3-develop` branches, this variable can be set to `'^v3-develop\|$^v2-develop$'`
+| SAFE_TEST                 |X          | false                                                                                 | Boolean on whether to run the build pipeline as a test before actually deploying, when set to \"true\" the build will not publish or deploy and artifacts.|
```
Comment (CorriganRJ) [inline README.md:27]: Would it be clearer to use true / false instead of "X" to indicate whether a variable is preloaded?

File: README.md [hunk 2]
```diff
@@ -118,11 +122,13 @@ Publishes a SNAPSHOT jar whenever a feature branch is merged into the project's
 
 | Variable                	| Default Value                                                        	| Description                                                                                                                                                                               	|
 |-------------------------	|----------------------------------------------------------------------	|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------   |
-| STANDARD_GRADLE_FLAGS   	| -s --no-daemon -PnoMavenLocal --refresh-dependencies --console=plain 	| Default Gradle flags that will be appended to all Gradle commands                                                                                                                             |
+| STANDARD_GRADLE_FLAGS   	|           | -s --no-daemon -PnoMavenLocal --refresh-dependencies --console=plain  (-PsafeTest)	| Default Gradle flags that will be appended to all Gradle commands (Will include -PsafeTest when set)                                                                                        |
```
Comment (CorriganRJ) [inline README.md:125]: We should probably call out the `SAFE_TEST` variable here if we're going to mention the flag.

File: lib/gitlab/ci/templates/pipeline/GradleJavaPipeline.yml [hunk 3]
```diff
@@ -42,6 +49,9 @@ stages:
 
 workflow:
   rules:
+    - if: '$SAFE_TEST =~ /true/i'
```
Comment (CorriganRJ) [inline lib/gitlab/ci/templates/pipeline/GradleJavaPipeline.yml:52]: This is a clever approach. We'll definitely want to introduce more variables using this pattern in future iterations

**Review by CorriganRJ — APPROVED** [review]

File: lib/gitlab/ci/templates/pipeline/GradleJavaPipeline.yml (hunk 3 above)
Comment (AndrewTrow) [reply lib/gitlab/ci/templates/pipeline/GradleJavaPipeline.yml:52]: Thanks!

**Pull request merged**